### PR TITLE
test: Add targets to build server main lib with limited list of extensions

### LINKED ISSUE #6736: //test/integration/... requires compiling all extensions
I have noticed that when testing `//test/integration/...` it compiles all extensions. I think this is probably because there is some server test in there that also compiles all extensions, but I'm not sure.

Optimally, we should be able to run these tests without having to compile any extra extensions beyond potentially core ones like router, HCM, etc. This is useful on platforms that won't support all extensions.

cc @junr03

### PR DESCRIPTION
**Pull request by yanavlasov**
Commit Message:
Add targets to build server main lib with limited list of extensions

Additional Description:
This change is to limit the number of extensions needed to build tests in the //test/integration/... directory.
Risk Level: Low, test only
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Fixes #6736 

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: source/extensions/all_extensions.bzl [hunk 1]
```diff
@@ -14,3 +14,21 @@ def envoy_all_extensions(blacklist = []):
...
     # These extensions can be removed on a site specific basis.
     return [v for k, v in all_extensions.items() if not k in blacklist]
+
+# Core extensions needed to run Envoy's integration tests.
+_core_extensions = [
+    "envoy.access_loggers.file",
+    "envoy.filters.http.router",
+    "envoy.filters.http.health_check",
+    "envoy.filters.network.http_connection_manager",
+    "envoy.filters.network.redis_proxy",
```
Comment (yanavlasov) [inline source/extensions/all_extensions.bzl:24]: This is used in the config for the hot_restart test. I'm unsure if it is needed by the test, so perhaps this target can be removed.

Comment (mattklein123) [reply source/extensions/all_extensions.bzl:24]: +1 let's remove it.

Comment (yanavlasov) [reply source/extensions/all_extensions.bzl]: done

**Comment by yanavlasov** [comment]
Query returned expected dependencies in the `//source/extensions/...` universe
```
$ bazel query "allpaths(//test/integration:run_envoy_test, //source/extensions/...)"|grep extensions
Loading: 0 packages loaded
Loading: 0 packages loaded
Loading: 0 packages loaded
//source/exe:envoy_main_common_with_core_extensions_lib
//source/exe:envoy_common_with_core_extensions_lib
//source/extensions/filters/http/common:pass_through_filter_lib
//source/extensions/upstreams/http/generic:config
//source/extensions/upstreams/http/tcp:upstream_request_lib
//source/extensions/upstreams/http/http:upstream_request_lib
//source/extensions/filters/listener:well_known_names
//source/extensions/transport_sockets/raw_buffer:config
//source/extensions/stat_sinks/statsd:config
//source/extensions/stat_sinks/common/statsd:statsd_lib
//source/extensions/stat_sinks:well_known_names
//source/extensions/filters/network/http_connection_manager:config
//source/extensions/filters/network/common:factory_base_lib
//source/extensions/filters/http/router:config
//source/extensions/filters/http/on_demand:on_demand_update_lib
//source/extensions/filters/http/health_check:config
//source/extensions/filters/http/health_check:health_check_lib
//source/extensions/filters/http/common:factory_base_lib
//source/extensions/access_loggers/file:config
//source/extensions/access_loggers/file:file_access_log_lib
//source/extensions/access_loggers/common:access_log_base
//source/extensions/access_loggers:well_known_names
//source/extensions/filters/network/common:utility_lib
//source/extensions/filters/network:well_known_names
//source/extensions/transport_sockets:well_known_names
//source/extensions/clusters:well_known_names
//source/extensions/common/proxy_protocol:proxy_protocol_header_lib
//source/extensions/filters/http/common:utility_lib
//source/extensions/filters/http:well_known_names
//source/extensions/common:utility_lib
$
```

**Comment by mattklein123** [comment]
Cool looks like needs a master merge.

/wait

**Pull request merged**